File: algorithms/javaalgo/src/test/java/programmers/level1/secretmap/SecretMapTest.java
package programmers.level1.secretmap;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

public class SecretMapTest {

    @Test
    void test1() {
        int n = 5;
        int[] arr1 = {9, 20, 28, 18, 11};
        int[] arr2 = {30, 1, 21, 17, 28};

        assertThat(solution(n, arr1, arr2)).isEqualTo(new String[]{
                "#####",
                "# # #",
                "### #",
                "#  ##",
                "#####"
            }
        );
    }

    @Test
    void test2() {
        int n = 6;
        int[] arr1 = {46, 33, 33, 22, 31, 50};
        int[] arr2 = {27, 56, 19, 14, 14, 10};

        assertThat(solution(n, arr1, arr2)).isEqualTo(new String[]{
                "######",
                "###  #",
                "##  ##",
                " #### ",
                " #####",
                "### # "
            }
        );
    }

    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];

        for (int i = 0; i < n; i++) {
            String map = Integer.toBinaryString(arr1[i] | arr2[i]);
            while (map.length() < n) {
                map = "0" + map;
            }
            map = map.replaceAll("0", " ");
            map = map.replaceAll("1", "#");
            answer[i] = map;
        }

        return answer;
    }
}

File: algorithms/javascriptalgo/programmers/level1/비밀_지도/secretMap.test.js
test("test", () => {
  const n1 = 5;
  const arr1 = [9, 20, 28, 18, 11];
  const arr2 = [30, 1, 21, 17, 28];
  const output1 = ["#####", "# # #", "### #", "#  ##", "#####"];

  const n2 = 6;
  const arr3 = [46, 33, 33, 22, 31, 50];
  const arr4 = [27, 56, 19, 14, 14, 10];
  const output2 = ["######", "###  #", "##  ##", " #### ", " #####", "### # "];

  expect(solution(n1, arr1, arr2)).toEqual(output1);
  expect(solution(n2, arr3, arr4)).toEqual(output2);
});

function solution(n, arr1, arr2) {
  const answer = [];
  for (let i = 0; i < n; i++) {
    let secret = (arr1[i] | arr2[i]).toString(2);
    while (secret.length < n) {
      secret = "0" + secret;
    }
    secret = secret.replaceAll("1", "#").replaceAll("0", " ");
    answer.push(secret);
  }

  return answer;
}
